### ISSUE TITLE
[Assert] Entering the Game Mode when the SlideAlongAxisBasedOnAngle in the Camera Rig is set results in an assert

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#3959]
**Describe the bug**
It has been noticed that entering the Game Mode when the SlideAlongAxisBasedOnAngle in the Camera Rig is set results in an assert.

**To Reproduce**
1. Create an entity.
2. Add Camera Rig component to it.
3. In the Look-at Behavior add SlideAlongAxisBasedOnAngle setting.
4. Enter the Game Mode.

**Expected behavior**
The Game Mode launches without issues.

**Desktop/Device:**
Device: PC
OS: Windows
Version 10
CPU Intel I5-9600k
GPU RTX 2060
Memory 16GB

**Additional context**

![assert52](https://user-images.githubusercontent.com/86952009/132332032-80a5b86f-a26c-43e2-aaed-a4a09bae0236.JPG)
```
File: C:/O3DE/o3de/Gems/StartingPointCamera/Code/Source/StartingPointCamera/StartingPointCameraUtilities.cpp: 
Condition:'void __cdecl Camera::MaskComponentFromNormalizedVector(class AZ::Vector3 &,enum Camera::VectorComponentType)'
MaskComponentFromNormalizedVector: VectorComponentType - unexpected value
Line: 52
```
https://user-images.githubusercontent.com/86952009/132332082-0e5fe832-fbf4-4cf4-ae8f-269f30fc72b0.mp4

**Comment by hultonha** [comment]
@monroegm A lot of these issues are related to the Camera Component which is not really to do with the feature/viewport work... I'm not certain what team owns StartingPointCamera but it would be good to get clarity on it as I'm not sure all these bugs belong to viewport 😝

**Comment by AMZN-Igarri** [comment]
This method basically 0 outs a vector component and the re-normalizes it so but I see some inconsistencies from an UX perspective.
Right now the user can either choose just one vector component (x, y or z) or choose none of them which to me seems a bit weird, why I cannot choose x and y and ignore z for example? I came across two possible solutions.|
1 - Remove the "None" option and make the user choose one component
2 - Make a checkbox, one for each vector component, and allow the user to ignore (or not) different combinations.

I personally prefer the second option but let me know your thoughts on that.

**Comment by amzn-leenguy** [comment]
I agree with you that the component card is using a "dropdown" for a multiple choice option. You are correct. This should be handled with checkboxes. And I feel like it's strange that we are asking the user to "ignore" rather than what is active or not active. (But I dont want to change it too drastically now)

Let's run with option 2 here for "Vector Component to Ignore" to have 3 check boxes in replacement for the dropdown for [  ] X      [  ] Y       [  ]  Z. And by default they are all unchecked. 

In regards to none,  we have a toggle that can disable a row of property data. @yuyihsu has used this pattern before. I can follow up with her and find another component card that uses it so you can copy it. But for now, by default none of the check boxes are ticked, so that is technically starting on none. And i will follow up later on if we need an option to allow the user to disable the whole row with a toggle and which component has an example of this.

**Comment by AMZN-Igarri** [comment]
I followed the PhysXRigidBodyComponent pattern for Linear and Angular Axis Locking and I ended up with this.
Everything is up in the PR.
![CameraRig](https://user-images.githubusercontent.com/82394219/140914878-3c41bede-5831-41e6-a058-d2898a97a42e.png)

**Comment by hultonha** [comment]
Closed by https://github.com/o3de/o3de/pull/5410

**Comment by LB-BartoszDudzinski** [comment]
Verified as fixed in Development (3791149) and Stabilization_2110 (7031147) branches.
